### PR TITLE
Add typing and pre-commit hooks

### PR DESCRIPTION
**Pull request by greschd**
Follow-up to #2, but depends on #3 (and a new "old" version being published to PyPI first).

Added CI for Python 2.7 and 3.5 where the module is installed from PyPI, but running the current tests. The reason for this is that we want to be sure the "graceful fallback" to installing old versions works as intended.

Once we're reasonably sure this works (a.k.a. with the next major change / release), we can probably drop these franken-tests.

While we're at it, I also added CI jobs which install the package through ``python setup.py sdist; pip install dist/...``, to make sure the MANIFEST file is complete.

### CODE REVIEW

**Comment by mskoenz** [comment]
Conflicts due to #3 need resolving

**Comment by greschd** [comment]
Done, but I'm still looking into whether the pre-commit hooks can be made nicer.

**Comment by mskoenz** [comment]
Ok, ping me when ready for review

**Comment by greschd** [comment]
Alright, good to go.

I've replaced the "local" pre-commit hooks with the mirrors. The difference is that "local" hooks use the current virtualenv, while the mirrors create their own. Since hooks with ``python`` language attribute [install the package in the venv](https://pre-commit.com/#python), that should still work as expected (notably, ``pylint`` needs the package / deps). 

I vaguely remember there being issues with different versions of ``pylint`` being used, but these days it seems ``prospector`` pins its dependencies, so that should also be fine.

Thanks for making me look into how pre-commit _should_ be used 😅

**Comment by greschd** [comment]
Note that the package will still install on py27 / 35 _locally_, but when installing from PyPI you will get the right version. To test I've uploaded a ``dev`` version (install with ``--pre`` flag).

File: README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# fsc.export
```
Comment (greschd) [inline README.md:1]: I moved the ``README.md`` (previously just ``README``) from the ``doc`` subfolder to the top-level, to have a minimal description on github. @mskoenz do you know if there was a strong reason why we put it in ``doc`` initially?

Comment (mskoenz) [reply README.md:1]: No reason comes to mind right now, change it and we'll revert if we remember ^^

Comment (greschd) [reply README.md:1]: 😂

**Comment by mskoenz** [comment]
py37: I installed `pip install fsc.export --pre`which tells me:
```
Successfully installed fsc.export-1.2.0.dev1
```
inspecting the version with:
```
from fsc import export
print(export.__version__)
```
yields
`1.2.0.dev1`

py35/27: `pip install fsc.export --pre` falls back to:
```
Successfully installed fsc.export-1.1.1
```

**Review by mskoenz — APPROVED** [review]
Works as intended:

* This version fails with appropriate error when trying to install with py27 or py35
* It works with py36 and py37, precommit hooks install nicely and pass on all files
* Installing --pre from pip gracefully falls back to 1.1.1 for py27 and py35

**Comment by mskoenz** [comment]
@greschd delete branch at your convenience

**Pull request merged**